### ISSUE TITLE
`plot_embeddings` with `image_sprites` from uri

### ISSUE THREAD

**Issue by abduhbm** [jina-ai/docarray#64]
**To produce:**

```python
from docarray import DocumentArray, Document
import torchvision

def preproc(d: Document):
    return (d.load_uri_to_image_tensor()  
             .set_image_tensor_shape((200, 200))  
             .set_image_tensor_normalization() 
             .set_image_tensor_channel_axis(-1, 0)) 

left_da = DocumentArray.from_files('left/*.jpg')

left_da.apply(preproc)

model = torchvision.models.resnet50(pretrained=True)  
left_da.embed(model)  

left_da.plot_embeddings(image_sprites=True)
```

Error:
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Input In [1], in <module>
     14 model = torchvision.models.resnet50(pretrained=True)  
     15 left_da.embed(model)  
---> 17 left_da.plot_embeddings(image_sprites=True)

File ~/workplaces/docarray-env/docarray/docarray/array/mixins/plot.py:152, in PlotMixin.plot_embeddings(self, title, path, image_sprites, min_image_size, channel_axis, start_server, host, port)
    140     if len(self) > max_docs:
    141         warnings.warn(
    142             f'''
    143             {self!r} has more than {max_docs} elements, which is the maximum number of image sprites can support. 
   (...)
    149             '''
    150         )
--> 152     self.plot_image_sprites(
    153         os.path.join(path, sprite_fn),
    154         canvas_size=canvas_size,
    155         min_size=min_image_size,
    156         channel_axis=channel_axis,
    157     )
    159 self.save_embeddings_csv(os.path.join(path, emb_fn), delimiter='\t')
    161 _exclude_fields = ('embedding', 'tensor', 'scores')

File ~/workplaces/docarray-env/docarray/docarray/array/mixins/plot.py:331, in PlotMixin.plot_image_sprites(self, output, canvas_size, min_size, channel_axis)
    329 row_id = floor(img_id / img_per_row)
    330 col_id = img_id % img_per_row
--> 331 sprite_img[
    332     (row_id * img_size) : ((row_id + 1) * img_size),
    333     (col_id * img_size) : ((col_id + 1) * img_size),
    334 ] = _d.tensor
    336 img_id += 1
    337 if img_id >= max_num_img:

ValueError: could not broadcast input array from shape (16,16,200) into shape (16,16,3)
```

**Suggested fix:** 
passing `use_uri` flag to `plot_embeddings` to allow visualizing the dots using `uri` attribute. I would be happy to raise a PR for this.

**Comment by abduhbm** [comment]
the broadcast issue can be resolved by checking the channel axis using `channel_axis=0` but this will not visualize the original images.

**Comment by JoanFM** [comment]
> the broadcast issue can be resolved by checking the channel axis using `channel_axis=0` but this will not visualize the original images.

maybe u can transform the axis back before plotting? I guess the `stacking` is only needed at `embedding` time

**Comment by abduhbm** [comment]
> maybe u can transform the axis back before plotting? I guess the `stacking` is only needed at `embedding` time

Changing the axis will work , however , it would be nice to have an option to show the original images when projecting embeddings without processing them again.

**Comment by JoanFM** [comment]
More than the original, you expect plot_embeddings to force some axis as the channel one?

**Comment by abduhbm** [comment]
> More than the original, you expect plot_embeddings to force some axis as the channel one?


just a flag to force loading from uri and setting the channel axis:
https://github.com/jina-ai/docarray/blob/79c13a09980c97c9b4fcf9857566ff2a46c95c83/docarray/array/mixins/plot.py#L321

```python
if use_uri or _d.content_type != 'tensor':
    _d.load_uri_to_image_tensor()
    channel_axis = -1
```

**Comment by numb3r3** [comment]
another possible approach for this problem. As the applied `preproc` function is only for model inference. Can we pass the `prepoc` function to  `. embed()`:

```python
def preproc(d: Document)
    ....
    return d

da.embed(model, preproc_fn=preproc)
```
Then, the output (copy of the origin DA) of `preproc` can then be fed into the model for inference, and will not change the value of origin DA. 

With this change, the user can choose which method to use regarding context. If the user wants to store the updates, then apply `preproc` before `.embed()`. If the user don't want keep the updates, then apply the `preproc` within the `.embed()`

**Comment by hanxiao** [comment]
@abduhbm make sense solved in #66 roll out soon

**Comment by abduhbm** [comment]
Thanks all for your support.
